src/lib/rank_utils.ts

```diff
@@ -22,9 +22,74 @@ interface IRankInfo {
     label: string;
 }
 
-export const MaxRank: number = 36;
+class Rating {
+    unset:boolean;
+    rating:number;
+    deviation:number;
+    volatility:number;
+    rank:number;
+    rank_label:string;
+    partial_rank:number;
+    partial_rank_label:string;
+    rank_deviation_labels:Array<string>;
+    rank_deviation:number;
+}
+
+export const MaxRank: number = 39;
+
+
+const MIN_RATING = 100;
+const MAX_RATING = 6000;
+
+export function rank_to_rating(rank:number) {
+    return 900 * Math.exp(0.032 * rank);
+}
+
+export function rating_to_rank(rating:number) {
+    return Math.log(Math.min(MAX_RATING, Math.max(MIN_RATING, rating)) / 900.0) / 0.032;
+}
+
+export function get_handicap_adjustment(rating:number, handicap:number):number {
+    return rank_to_rating(rating_to_rank(rating) + handicap) - rating;
```

Comment:
Alrighty I ran the numbers with Glicko-2

`H`
```
     	        1                2                3                4                5                6                7                8                9        
30k  	     38.5 [   26]      0.0 [    5]    100.0 [    3]      0.0 [    2]    100.0 [    1]     50.0 [    2]      0.0 [    0]      0.0 [    2]      0.0 [    2]
29k  	     46.9 [   32]     50.0 [    8]     50.0 [    2]      0.0 [    3]    100.0 [    1]      0.0 [    1]      0.0 [    0]      0.0 [    0]     50.0 [    2]
28k  	     46.9 [   49]     50.0 [    4]     33.3 [    3]      0.0 [    2]     50.0 [    2]      0.0 [    2]      0.0 [    1]      0.0 [    0]     25.0 [    4]
27k  	     37.9 [   66]     30.0 [   10]     16.7 [    6]     33.3 [    3]     50.0 [    4]      0.0 [    3]     33.3 [    3]     33.3 [    3]     33.3 [    6]
26k  	     32.4 [  105]     40.0 [   15]     14.3 [    7]     28.6 [    7]     33.3 [    3]      0.0 [    2]     50.0 [    2]     25.0 [    4]     14.3 [    7]
25k  	     46.1 [  128]     54.5 [   33]     36.8 [   19]     29.4 [   17]     80.0 [   10]     14.3 [    7]      0.0 [    6]      0.0 [    2]     33.3 [    6]
24k  	     39.4 [  175]     41.5 [   41]     41.7 [   24]     27.3 [   11]     35.3 [   17]      0.0 [    7]      0.0 [   11]      0.0 [    4]     13.3 [   15]
23k  	     46.9 [  211]     32.8 [   61]     31.6 [   38]     15.4 [   26]     35.7 [   14]     30.0 [   10]      0.0 [   11]     41.7 [   12]      6.2 [   16]
22k  	     41.8 [  340]     34.1 [   85]     39.1 [   46]     46.7 [   30]     21.7 [   23]     45.0 [   20]      6.7 [   15]      0.0 [   12]     24.0 [   25]
21k  	     41.9 [  418]     39.1 [  133]     46.0 [   63]     32.7 [   55]     38.2 [   34]     35.7 [   42]     35.3 [   17]     16.1 [   31]     36.4 [   33]
20k  	     44.1 [  515]     39.6 [  182]     42.0 [  100]     27.9 [   61]     43.6 [   39]     25.5 [   47]     30.3 [   33]     16.1 [   31]     22.9 [   35]
19k  	     41.5 [  702]     35.2 [  253]     35.8 [  148]     34.3 [  105]     27.9 [   61]     29.1 [   79]     23.7 [   59]     32.7 [   52]     27.3 [   55]
18k  	     42.5 [  870]     44.6 [  332]     32.9 [  222]     30.8 [  156]     40.5 [  116]     31.1 [  103]     24.3 [   74]     26.4 [   53]     37.1 [   70]
17k  	     41.3 [ 1186]     40.4 [  488]     37.9 [  282]     43.0 [  214]     37.1 [  151]     35.9 [  128]     32.9 [   79]     26.8 [   71]     31.8 [   66]
16k  	     46.5 [ 1485]     44.4 [  581]     37.3 [  370]     38.8 [  258]     33.5 [  185]     32.4 [  145]     32.0 [  103]     36.1 [   83]     21.1 [   76]
15k  	     41.4 [ 1942]     40.6 [  950]     36.0 [  598]     35.7 [  431]     40.2 [  311]     35.2 [  233]     31.1 [  132]     35.8 [  106]     31.9 [   94]
14k  	     42.8 [ 2060]     39.9 [  989]     41.7 [  652]     38.0 [  458]     42.5 [  365]     44.0 [  182]     28.7 [  115]     33.7 [   92]     36.4 [   66]
13k  	     44.5 [ 2472]     42.0 [ 1277]     42.1 [  877]     43.8 [  633]     38.7 [  408]     43.6 [  195]     37.2 [  148]     31.4 [  105]     36.1 [   61]
12k  	     44.5 [ 2665]     44.9 [ 1542]     41.1 [  973]     42.5 [  595]     38.8 [  340]     35.1 [  202]     32.1 [  137]     29.3 [   82]     28.2 [   71]
11k  	     42.5 [ 2966]     44.8 [ 1766]     42.3 [  955]     44.5 [  542]     47.3 [  319]     43.5 [  193]     40.7 [  113]     32.9 [   79]     37.9 [   66]
10k  	     42.5 [ 3303]     44.6 [ 1787]     43.8 [  906]     41.0 [  503]     37.2 [  274]     44.7 [  170]     28.7 [  115]     41.0 [   61]     32.6 [   43]
9k   	     44.2 [ 3372]     44.4 [ 1682]     39.0 [  871]     41.5 [  467]     37.2 [  269]     42.6 [  183]     34.4 [  125]     38.5 [   52]     39.6 [   48]
8k   	     45.5 [ 2883]     44.6 [ 1473]     39.3 [  717]     39.6 [  376]     34.0 [  197]     38.3 [  149]     32.9 [   85]     34.4 [   61]     29.2 [   65]
7k   	     42.1 [ 2196]     42.3 [ 1231]     43.5 [  593]     41.4 [  280]     46.4 [  192]     40.0 [  115]     38.4 [   73]     36.5 [   74]     29.1 [   55]
6k   	     43.6 [ 1942]     45.6 [ 1008]     41.9 [  475]     43.4 [  256]     42.4 [  144]     40.6 [  101]     23.1 [   78]     34.8 [   46]     43.2 [   37]
5k   	     42.6 [ 1589]     42.9 [  812]     38.3 [  447]     39.6 [  225]     38.1 [  134]     44.6 [  112]     38.9 [   72]     31.2 [   32]     33.3 [   18]
4k   	     44.1 [ 1362]     41.4 [  688]     39.1 [  340]     38.5 [  231]     43.5 [  131]     42.9 [   63]     41.0 [   61]     47.1 [   17]     17.4 [   23]
3k   	     40.9 [ 1081]     36.5 [  504]     40.8 [  343]     36.3 [  190]     35.7 [  126]     44.4 [   54]     63.3 [   30]     40.0 [   10]     42.9 [   35]
2k   	     41.5 [  856]     38.2 [  482]     36.1 [  310]     39.9 [  178]     50.0 [  100]     51.1 [   45]     45.5 [   22]     19.0 [   21]     28.0 [   25]
1k   	     38.5 [  644]     40.0 [  380]     40.1 [  202]     42.7 [  117]     47.3 [   74]     36.0 [   25]     46.2 [   26]     32.4 [   37]     10.5 [   19]
1d   	     39.4 [  449]     45.7 [  197]     44.4 [  117]     52.2 [   90]     46.5 [   43]     25.0 [   24]     26.3 [   19]     21.7 [   23]      0.0 [    0]
2d   	     39.3 [  257]     33.3 [  120]     45.3 [   86]     39.5 [   43]     44.1 [   34]     31.8 [   22]     20.0 [    5]     50.0 [    2]      0.0 [    0]
3d   	     37.4 [  147]     39.4 [   94]     66.7 [   21]     48.4 [   31]     44.0 [   25]     25.0 [    4]      0.0 [    0]    100.0 [    1]      0.0 [    0]
4d   	     39.5 [   76]     59.4 [   32]     35.7 [   28]     50.0 [   14]     28.6 [    7]     50.0 [    2]      0.0 [    0]      0.0 [    0]      0.0 [    0]
5d   	     34.8 [   69]     66.7 [   15]     38.7 [   31]     80.0 [    5]      0.0 [    1]      0.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]
6d   	     49.2 [   63]     55.9 [   34]     47.6 [   21]      0.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
7d   	     56.2 [   32]     53.3 [   15]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
8d   	     43.3 [   30]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
9d   	     28.6 [    7]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
10d  	      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]



  30k-  27k	     42.0 [ 3888]     29.2 [   24]     37.5 [   16]     22.2 [    9]     28.6 [   14]
  26k-  23k	     43.8 [14066]     29.5 [  149]     34.3 [   70]     13.5 [   52]     13.6 [   44]
  22k-  19k	     44.4 [46661]     37.3 [  608]     32.2 [  345]     23.2 [  250]     27.7 [  148]
  18k-  15k	    43.6 [140314]     36.6 [ 2531]     36.2 [ 1372]     31.2 [  701]     30.4 [  306]
  14k-  11k	    46.5 [195456]     42.0 [ 5685]     41.6 [ 2204]     33.5 [  871]     34.5 [  264]
  10k-   7k	    47.5 [197213]     41.2 [ 4713]     39.7 [ 1549]     34.8 [  646]     32.2 [  211]
   6k-   3k	    48.0 [155017]     39.9 [ 2507]     41.2 [  865]     37.0 [  346]     36.3 [  113]
   2k-   2d	     47.8 [66536]     41.0 [ 1143]     45.0 [  367]     32.3 [  155]     20.5 [   44]
   3d-   6d	     48.4 [ 8008]     47.4 [  152]     37.5 [   40]    100.0 [    1]      0.0 [    0]
   7d-  10d	     47.5 [  575]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]

Overall: 46.2%
```

`H-0.5`
```
     	        1                2                3                4                5                6                7                8                9        
30k  	     33.3 [   21]     50.0 [    4]    100.0 [    2]     50.0 [    2]    100.0 [    1]     50.0 [    2]      0.0 [    0]      0.0 [    2]      0.0 [    1]
29k  	     43.5 [   23]     25.0 [    8]      0.0 [    0]    100.0 [    1]    100.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]    100.0 [    2]
28k  	     60.5 [   43]     50.0 [    6]     20.0 [    5]      0.0 [    4]      0.0 [    1]      0.0 [    2]      0.0 [    1]      0.0 [    1]     33.3 [    3]
27k  	     31.3 [   67]     22.2 [    9]     42.9 [    7]      0.0 [    1]     33.3 [    3]      0.0 [    3]      0.0 [    1]     33.3 [    3]     20.0 [    5]
26k  	     35.6 [   87]     40.0 [   15]     12.5 [    8]     25.0 [    8]     20.0 [    5]     25.0 [    4]      0.0 [    1]      0.0 [    1]     14.3 [    7]
25k  	     48.9 [  131]     40.0 [   35]     41.2 [   17]     14.3 [   14]     44.4 [    9]     18.2 [   11]     14.3 [    7]     28.6 [    7]     12.5 [    8]
24k  	     41.8 [  165]     35.1 [   37]     61.1 [   18]     28.6 [   14]     46.2 [   13]      0.0 [    9]      0.0 [    8]      0.0 [    4]     23.5 [   17]
23k  	     43.8 [  210]     35.2 [   54]     23.8 [   42]     17.6 [   17]     26.3 [   19]     50.0 [    6]      0.0 [    8]     20.0 [   15]      0.0 [   15]
22k  	     39.4 [  327]     39.6 [   91]     31.6 [   38]     25.7 [   35]     27.3 [   22]     43.5 [   23]      0.0 [   19]      6.7 [   15]     22.2 [   27]
21k  	     38.2 [  434]     37.7 [  138]     46.4 [   69]     41.8 [   55]     37.1 [   35]     30.3 [   33]     37.5 [   16]     21.4 [   28]     38.2 [   34]
20k  	     45.8 [  496]     43.5 [  186]     43.3 [   97]     31.3 [   67]     28.6 [   35]     34.5 [   55]     20.0 [   30]     14.7 [   34]     16.7 [   42]
19k  	     40.6 [  712]     33.9 [  248]     34.6 [  130]     34.8 [   92]     32.4 [   68]     26.9 [   78]     27.3 [   66]     22.9 [   48]     25.5 [   51]
18k  	     41.7 [  884]     41.8 [  347]     33.6 [  244]     32.5 [  163]     36.8 [  114]     33.9 [  109]     25.0 [   68]     27.1 [   59]     35.7 [   56]
17k  	     42.8 [ 1218]     41.2 [  473]     36.7 [  264]     42.2 [  225]     36.0 [  161]     29.8 [  141]     35.0 [   80]     25.0 [   72]     29.6 [   71]
16k  	     46.9 [ 1575]     43.0 [  635]     39.5 [  385]     38.7 [  266]     36.1 [  208]     31.9 [  160]     33.3 [  102]     30.9 [   68]     24.6 [   69]
15k  	     41.4 [ 1941]     41.6 [  973]     35.8 [  587]     36.9 [  431]     38.5 [  325]     38.2 [  241]     28.3 [  145]     33.0 [  109]     28.6 [   91]
14k  	     41.5 [ 2088]     39.9 [ 1021]     40.3 [  680]     39.1 [  486]     39.6 [  364]     38.1 [  194]     37.8 [  111]     37.2 [   94]     50.0 [   60]
13k  	     43.4 [ 2507]     41.5 [ 1322]     40.3 [  943]     42.7 [  618]     42.2 [  410]     40.1 [  197]     30.4 [  138]     30.1 [  103]     32.3 [   62]
12k  	     44.7 [ 2827]     44.5 [ 1639]     41.2 [ 1021]     41.1 [  620]     38.8 [  348]     36.8 [  209]     33.6 [  131]     33.0 [   91]     23.3 [   73]
11k  	     42.0 [ 3133]     43.2 [ 1853]     41.3 [ 1008]     42.1 [  560]     45.6 [  307]     42.0 [  193]     39.2 [  120]     30.5 [   82]     40.5 [   74]
10k  	     42.3 [ 3458]     44.0 [ 1861]     43.0 [  965]     39.8 [  500]     36.9 [  287]     46.2 [  160]     28.0 [  107]     32.1 [   56]     32.6 [   46]
9k   	     43.7 [ 3466]     42.9 [ 1688]     38.8 [  884]     37.7 [  462]     33.8 [  266]     38.7 [  173]     32.5 [  114]     41.8 [   55]     26.9 [   52]
8k   	     45.1 [ 2834]     45.3 [ 1459]     39.0 [  712]     41.0 [  373]     34.7 [  196]     39.7 [  131]     30.9 [   81]     30.8 [   65]     26.4 [   72]
7k   	     42.5 [ 2247]     41.0 [ 1208]     44.1 [  574]     44.3 [  287]     45.9 [  183]     37.1 [  116]     32.4 [   74]     41.4 [   70]     26.7 [   45]
6k   	     43.4 [ 1960]     45.5 [  970]     39.3 [  496]     38.2 [  249]     40.8 [  152]     41.6 [  113]     30.4 [   92]     36.2 [   47]     27.8 [   18]
5k   	     42.3 [ 1595]     39.8 [  822]     40.4 [  423]     39.3 [  239]     38.7 [  142]     43.7 [  103]     36.4 [   55]     50.0 [   20]     25.0 [   12]
4k   	     43.5 [ 1355]     39.6 [  677]     37.1 [  348]     37.8 [  246]     46.8 [  124]     49.1 [   53]     47.5 [   61]     36.0 [   25]     22.2 [   36]
3k   	     40.6 [ 1099]     36.0 [  528]     40.5 [  380]     36.2 [  218]     41.7 [  127]     46.7 [   45]     44.0 [   25]     23.1 [   13]     33.3 [   42]
2k   	     39.1 [  857]     35.5 [  519]     36.8 [  307]     35.8 [  165]     43.2 [   81]     43.2 [   37]     30.6 [   36]     17.5 [   40]     30.0 [   30]
1k   	     37.8 [  588]     39.5 [  314]     34.1 [  179]     47.9 [  121]     49.2 [   59]     37.5 [   24]     27.6 [   29]     30.0 [   30]     20.0 [    5]
1d   	     39.2 [  421]     44.0 [  166]     43.1 [  123]     43.6 [   78]     40.0 [   30]     23.8 [   21]     28.6 [   21]     15.4 [   13]      0.0 [    0]
2d   	     35.1 [  208]     38.5 [  109]     39.6 [   53]     41.3 [   46]     37.0 [   27]     30.8 [   13]    100.0 [    2]    100.0 [    1]      0.0 [    0]
3d   	     38.3 [  115]     43.8 [   64]     57.1 [   21]     37.9 [   29]     44.4 [   18]     33.3 [    3]      0.0 [    1]      0.0 [    0]      0.0 [    0]
4d   	     29.4 [   85]     46.7 [   15]     33.3 [   39]     42.1 [   19]     50.0 [    2]     50.0 [    2]      0.0 [    0]      0.0 [    0]      0.0 [    0]
5d   	     36.5 [   63]     66.7 [   12]     51.9 [   27]      0.0 [    0]      0.0 [    0]      0.0 [    1]      0.0 [    0]      0.0 [    0]      0.0 [    0]
6d   	     66.7 [   36]     38.5 [   26]     50.0 [    2]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
7d   	     57.4 [   47]     66.7 [    6]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
8d   	     50.0 [    8]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
9d   	      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]
10d  	      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]



  30k-  27k	     42.0 [ 3571]     36.4 [   22]     30.8 [   13]     12.5 [    8]     36.4 [   11]
  26k-  23k	     43.8 [13549]     29.0 [  138]     28.9 [   76]     11.8 [   51]     12.8 [   47]
  22k-  19k	     44.3 [45798]     37.0 [  583]     31.8 [  349]     20.7 [  256]     25.3 [  154]
  18k-  15k	    43.6 [140453]     37.1 [ 2565]     35.8 [ 1459]     30.0 [  703]     29.3 [  287]
  14k-  11k	    46.4 [200288]     41.0 [ 5936]     40.6 [ 2222]     34.0 [  870]     36.1 [  269]
  10k-   7k	    47.3 [203208]     40.8 [ 4757]     38.6 [ 1512]     33.1 [  622]     27.9 [  215]
   6k-   3k	    47.9 [157860]     38.9 [ 2599]     42.7 [  859]     37.6 [  338]     27.8 [  108]
   2k-   2d	     47.6 [63647]     39.0 [ 1072]     41.1 [  292]     26.7 [  172]     28.6 [   35]
   3d-   6d	     48.5 [ 6899]     43.1 [  137]     42.3 [   26]      0.0 [    1]      0.0 [    0]
   7d-  10d	     48.8 [  449]      0.0 [    0]      0.0 [    0]      0.0 [    0]      0.0 [    0]

Overall: 46.1%
```

So it looks like it really doesn't matter much, but `H` seemed to "win" out by `0.1%`, though that's close enough to be equal in this case I think. Never the less, there doesn't seem to be any reason to change it to `H-0.5` as far as I can tell from this.